internal/pkg/list/list.go

```diff
@@ -0,0 +1,170 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package list
+
+import (
+	"encoding/json"
+	"fmt"
+	"io"
+	"math"
+	"strings"
+	"text/tabwriter"
+
+	"github.com/aws/copilot-cli/internal/pkg/config"
+)
+
+const (
+	// Display settings.
+	minCellWidth           = 20  // minimum number of characters in a table's cell.
+	tabWidth               = 4   // number of characters in between columns.
+	cellPaddingWidth       = 2   // number of padding characters added by default to a cell.
+	paddingChar            = ' ' // character in between columns.
+	noAdditionalFormatting = 0
+
+	// Workload types.
+	jobWorkloadType = "job"
+	svcWorkloadType = "service"
+)
+
+// Store wraps the methods required for interacting with config stores.
+type Store interface {
+	GetApplication(appName string) (*config.Application, error)
+	ListJobs(appName string) ([]*config.Workload, error)
+	ListServices(appName string) ([]*config.Workload, error)
+}
+
+// Workspace wraps the methods required to interact with a local workspace.
+type Workspace interface {
+	JobNames() ([]string, error)
+	ServiceNames() ([]string, error)
+}
+
+// JobListWriter holds all the metadata needed to
+type JobListWriter struct {
+	ShowLocalJobs bool
+	OutputJSON    bool
+
+	Store Store
+	Ws    Workspace
+	W     io.Writer
```

Comment:
one tiny nit:
```suggestion
	Out     io.Writer
```
And maybe some comments in the struct might make it easier to read.
```go
type  JobListWriter struct {
     // Configuration for how the output should be displayed.
     ShowLocalJobs bool
    OutputJSON bool

     Store Store // Client to retrieve application configuration.
     Ws Workspace // Client to retrieve local jobs.
     Out io.Writer // The writer where the output will be written.
}
```